### PR TITLE
Add marker type

### PR DESCRIPTION
**Pull request by nathancoliver**
I added marker specifications for marker symbol, size, fill color, and line color. I also added a default marker line width of 2.

### CODE REVIEW

File: dimes/dimes.py [hunk 1]
```diff
@@ -20,6 +20,10 @@ def __init__(
         dimension: Union[str, None] = None,
         color: Union[str, None] = None,
         line_type: Union[str, None] = None,
+        marker_symbol: Union[str, None] = None,
+        marker_size: Union[int, None] = None,
+        marker_line_color: Union[str, None] = None,
+        marker_fill_color: Union[str, None] = None,
```
Comment (nealkruis) [inline dimes/dimes.py:26]: This is getting a little cumbersome. I think we should move these inputs into a new class (maybe a `dataclass`?).

File: dimes/dimes.py [hunk 2]
```diff
@@ -132,12 +140,31 @@ def finalize_plot(self):
...
+                                    if all(
+                                        variables is None
+                                        for variables in (
+                                            time_series.marker_size,
+                                            time_series.marker_symbol,
+                                            time_series.marker_line_color,
+                                            time_series.marker_fill_color,
+                                        )
+                                    )
+                                    else "lines+markers",
```
Comment (nealkruis) [inline dimes/dimes.py:153]: This can be set up in the new class.

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: dimes/dimes.py (hunk 2 above)
Comment (nathancoliver) [reply dimes/dimes.py]: @nealkruis I created a new class called LineMode to solve this issue.

File: dimes/dimes.py (hunk 1 above)
Comment (nathancoliver) [reply dimes/dimes.py]: @nealkruis I moved these inputs into a dataclass called LineProperties.

File: dimes/dimes.py [hunk 3]
```diff
@@ -2,25 +2,52 @@
...
+                    marker_size,
+                    marker_symbol,
+                    marker_line_color,
+                    marker_fill_color,
+                )
+            ):
+                return "lines"
+            else:
+                return "lines+markers"
+
```
Comment (nealkruis) [inline dimes/dimes.py:42]: All of these variables are already data members of the LineProperties class. Why not add a method there instead?

**Review by nealkruis — CHANGES_REQUESTED** [review]

**Review by nealkruis — APPROVED** [review]

**Pull request merged**